server/src/main/java/com/google/musicanalysis/site/OAuthCallbackServlet.java

```diff
@@ -36,41 +52,36 @@ protected void doGet(HttpServletRequest req, HttpServletResponse res)
       return;
     }
 
-    var sessionOauthState = req.getSession().getAttribute("oauth:state");
+    var sessionOauthState =
+        (String) req.getSession().getAttribute("oauth-state-" + getServiceName());
```

Comment:
nit: remove some of the magic here and have a getStateSessionAttribute(). This will let you share some const `SPOTIFY_SESSION_ATTRUBUTE = "oauth-state-spotify"` between this getter and its setter.

Reply:
It also creates a hard link so that updating one forces you to update the other.